ow_power_system/include/PredictionHandler.h

```diff
@@ -0,0 +1,38 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef __PREDICTION_HANDLER_H__
+#define __PREDICTION_HANDLER_H__
+
+#include <vector>
+#include <map>
+
+#include "Messages/MessageBus.h"
+#include "Messages/ProgEventMessage.h"
+#include "Messages/ScalarMessage.h"
+#include "ModelBasedAsyncPrognoserBuilder.h"
+
+// The PredictionHandler class subscribes to the battery EoD event message and
+// prints each event as it is received.
+class PredictionHandler : public IMessageProcessor
+{
+public:
+  PredictionHandler(double& rul, double& soc, double& temp, MessageBus& bus,
+                    const std::string& src, int node_num);
+  ~PredictionHandler();
+  void processMessage(const std::shared_ptr<Message>& message) override;
+private:
+  double findMedian(std::vector<double> samples);
+
+  // References used by other classes to process EoD predictions.
+  double& m_rul_ref;
+  double& m_soc_ref;
+  double& m_temp_ref;
+  MessageBus& m_bus;
```

Comment:
This idiom (aggregation instead of composition, or having a copy of the objects instead of references) can be risky.  I need to look at the overall usage, but don't have time at the moment.

Reply:
I more or less had to implement it this way, because PredictionHandler needs to get its results into the variables of the PowerSystemPack object. PowerSystemPack is the object that handles and stores these variables, which also controls PredictionHandlers, so I think it's alright. I could be misusing it though still.